specification/containerregistry/data-plane/Azure.ContainerRegistry/preview/2019-08-15-preview/containerregistry.json

```diff
@@ -1,3091 +1,3091 @@
+              },
+              "readEnabled": {
+                  "type": "boolean",
+                  "description": "Read enabled",
+                  "x-ms-client-name": "canRead"
+              },
+              "teleportEnabled": {
+                  "type": "boolean",
+                  "description": "Enables Teleport functionality on new images in the repository improving Container startup performance"
+              }
           },
-          {
-            "changeableAttributes": {
-              "quarantineDetails": "quarantineDetails",
+          "example": {
               "readEnabled": true,
-              "quarantineState": "quarantineState",
               "listEnabled": true,
               "deleteEnabled": true,
-              "writeEnabled": true
-            },
-            "os": "os",
-            "digest": "digest",
-            "imageSize": 2401606,
-            "createdTime": "createdTime",
-            "mediaType": "mediaType",
-            "configMediaType": "configMediaType",
-            "lastUpdateTime": "lastUpdateTime",
-            "architecture": "architecture",
-            "tags": [
-              "tags",
-              "tags"
-            ]
+              "writeEnabled": true,
+              "teleportEnabled": true
           }
-        ]
-      }
-    },
-    "ManifestAttributes": {
-      "required": [
-        "manifest"
-      ],
-      "x-ms-client-name": "ArtifactManifestProperties",
-      "description": "Manifest attributes details",
-      "properties": {
-        "registry": {
-          "description": "Registry login server.",
-          "readOnly": true,
-          "type": "string",
-          "x-ms-client-name": "registryLoginServer"
-        },
-        "imageName": {
-          "description": "Repository name",
-          "type": "string",
-          "readOnly": true,
-          "x-ms-client-name": "repositoryName"
-        },
-        "manifest": {
-          "x-ms-client-flatten": true,
-          "readOnly": true,
-          "description": "Manifest attributes",
-          "$ref": "#/definitions/ManifestAttributesBase"
-        }
-      },
-      "example": {
-        "registry": "acrapi.azurecr-test.io",
-        "imageName": "nanoserver",
-        "manifest": {
-          "digest": "sha256:110d2b6c84592561338aa040b1b14b7ab81c2f9edbd564c2285dd7d70d777086",
-          "imageSize": 2401606,
-          "createdTime": "2018-09-06T06:17:20.9983915Z",
-          "lastUpdateTime": "2018-09-06T06:17:20.9983915Z",
-          "architecture": "amd64",
-          "os": "windows",
-          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
-          "configMediaType": "application/vnd.docker.container.image.v1+json",
-          "tags": [
-            "4.7.2-20180905-nanoserver-1803"
+      },
+      "Manifest": {
+          "discriminator": "mediaType",
+          "required": [
+              "mediaType",
+              "schemaVersion"
```

Comment:
schemaVersion although currently REQUIRED in OCI image spec, there are discussions to deprecate and remove it. I think the same is called out in the spec "This field MAY be removed in a future version of the specification."

Reply:
schemaVersion is the only field on the original Manifest type definition, and because of this, the only field inherited from it by all manifests.  Would it make sense to remove the Manifest base type?

Reply:
(I can also do this in a swagger transform for OciManifest)

Reply:
Yes, if there aren't any common shared properties, we can remove the base class.